src/anyio/_backends/_asyncio.py

```diff
@@ -219,7 +219,7 @@ def run(self, coro: Coroutine[T_Retval], *, context=None) -> T_Retval:
                 if self._interrupt_count > 0:
                     uncancel = getattr(task, "uncancel", None)
                     if uncancel is not None and uncancel() == 0:
-                        raise KeyboardInterrupt()
+                        raise KeyboardInterrupt from None
```

Comment:
Isn't this supposed to behave the same as asyncio.run? I want to see the context of the cancellation here

Reply:
Perhaps I should set the `__cause__` then rather than `__context__`?

Reply:
I'd be tempted to ignore the lint here, maybe even backport Runner and use that?

Reply:
Haven't I already backported the `Runner`, at the top of the module?

Reply:
I've reverted this particular change and added the #noqa.